### PR TITLE
Homework3&4

### PR DESCRIPTION
**Pull request by Mad-Ninja**
database implemented and unit test written

### CODE REVIEW

File: controllers/accounts.js [hunk 1]
```diff
@@ -1,37 +1,37 @@
-const { users } = require('../database/users.js');
+import User from '../models/users.js';
 
 function addAccount(req, res) {
-  users.push(req.body);
-  res.status(200).json({ message: 'Account created' });
+  // users.push(req.body);
```
Comment (devbarden) [inline controllers/accounts.js:4]: why its commented ?

File: controllers/accounts.js [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
 }
 
 function editAccount(req, res) {
-  const _id = req.params.id;
-  if (users[_id] == undefined || !users[_id].hasOwnProperty('id')) {
-    res.status(404).json({ message: 'Id is not exist' });
-  }
-  users[_id] = req.body;
-  res.status(200).json({ message: 'Account edited' });
+  // const _id = req.params.id;
```
Comment (devbarden) [inline controllers/accounts.js:9]: the same

File: controllers/users.js [hunk 3]
```diff
@@ -1,39 +1,57 @@
-const { users } = require('../database/users.js');
+import { ObjectId } from 'mongodb';
+import User from '../models/users.js';
 
-function addUser(req, res) {
-  users.push(req.body);
-  res.status(200).json({ message: 'User created' });
-}
+const mongoObjectId = function () {
+  let timestamp = (new Date().getTime() / 1000 | 0).toString(16);
```
Comment (devbarden) [inline controllers/users.js:5]: why 'let' ?

File: controllers/users.js [hunk 4]
```diff
@@ -1,39 +1,57 @@
-const { users } = require('../database/users.js');
+import { ObjectId } from 'mongodb';
+import User from '../models/users.js';
 
-function addUser(req, res) {
-  users.push(req.body);
-  res.status(200).json({ message: 'User created' });
-}
+const mongoObjectId = function () {
```
Comment (devbarden) [inline controllers/users.js:4]: if you need to generate id so please use uuid. https://www.npmjs.com/package/uuid

File: test/users.test.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import supertest from 'supertest';
```
Comment (devbarden) [inline test/users.test.js:1]: i see you added test. as I understand this is part of homework 4 ?

Comment (devbarden) [reply test/users.test.js:1]: if so then please rewrite your PR name that is homework 3 and 4

File: controllers/users.js (hunk 4 above)
Comment (Mad-Ninja) [reply controllers/users.js]: done

File: controllers/accounts.js (hunk 2 above)
Comment (Mad-Ninja) [reply controllers/accounts.js]: fixed

File: controllers/accounts.js (hunk 1 above)
Comment (Mad-Ninja) [reply controllers/accounts.js]: fixed

File: test/users.test.js (hunk 5 above)
Comment (Mad-Ninja) [reply test/users.test.js:1]: yes, it's 3&4 homework.  I renamed the PR :)

File: controllers/users.js (hunk 3 above)
Comment (Mad-Ninja) [reply controllers/users.js]: fixed)

**Comment by devbarden** [comment]
you can merge it if you want

**Review by devbarden — APPROVED** [review]

**Pull request merged**